src/transport.ts

```diff
@@ -1,16 +1,37 @@
-import {Connection} from '@libp2p/interface-connection';
-import {CreateListenerOptions, DialOptions, Listener, Transport} from '@libp2p/interface-transport'
-import {symbol} from '@libp2p/interface-transport'
-import {Multiaddr} from '@multiformats/multiaddr';
+import {createListener}        from './listener.js'
+import {toMultiaddrConnection} from './socket-to-conn.js'
+import {Components}         from '@libp2p/components'
+import {Connection}            from '@libp2p/interface-connection';
+import {CreateListenerOptions} from '@libp2p/interface-transport'
+import {Listener, Transport}   from '@libp2p/interface-transport'
+import {DialOptions}           from '@libp2p/interface-transport'
+import {symbol}                from '@libp2p/interface-transport'
+import {logger}                from '@libp2p/logger'
+import {Multiaddr}             from '@multiformats/multiaddr';
+import {WebRTCInitiator}       from '@libp2p/webrtc-peer'
+
+const log = logger('libp2p:webrtc:transport')
+const noop = () => {}
+
+export interface WebRTCDialOptions extends DialOptions {
+//   channelOptions?: WebRTCInitiatorInit
+}
+
+function ma2sdp(ma: Multiaddr): RTCSessionDescription {
+	return new RTCSessionDescription(JSON.parse("{TODO: \"" + ma + "\"}"));
+}
 
 export class WebRTCTransport implements Transport {
 
 	async dial(ma: Multiaddr, options: DialOptions): Promise<Connection> {
+		const rawConn = this._connect(ma, options)
+		const maConn = toMultiaddrConnection(rawConn, { remoteAddr: ma, signal: options.signal })
```

Comment:
we should be implementing the connection interface instead of `MultiaddrConnection`.